service/src/main/java/uk/gov/hmcts/reform/fpl/service/CMODocmosisTemplateDataGenerationService.java

```diff
@@ -92,8 +91,17 @@
         cmoTemplateData.put("representatives",
             getRepresentatives(caseData.getRespondents1(), applicantName, caseData.getSolicitor()));
 
-        // Populate with the next hearing booking, currently not captured
-        cmoTemplateData.putAll(commonCaseDataExtractionService.getHearingBookingData(null));
+        CaseManagementOrder order = draftCMOService.prepareCMO(caseData, getCaseManagementOrder(caseData));
+
+        if (order.getAction() != null && order.getAction().getNextHearingId() != null) {
+            List<Element<HearingBooking>> hearingBookings = caseData.getHearingDetails();
+            UUID nextHearingId = order.getAction().getNextHearingId();
+
+            HearingBooking nextHearing = hearingBookingService.getHearingBookingByUUID(hearingBookings, nextHearingId);
+            cmoTemplateData.putAll(commonCaseDataExtractionService.getHearingBookingData(nextHearing));
+        } else {
+            cmoTemplateData.putAll(commonCaseDataExtractionService.getHearingBookingData(null));
```

Comment:
`else` block can be removed and `cmoTemplateData.putAll` can be moved outside `if` block when we set `HearingBooking nextHearing = null` before it.